R/utilities-sensitivity-analysis.R

```diff
@@ -403,30 +386,78 @@ runPopulationSensitivityAnalysis <- function(structureSet, settings, logFolder =
+  filteredPkResultsList <- list()
+  for (n in seq_along(pkParameterNamesEachOutput)) {
+    op <- names(pkParameterNamesEachOutput)[n]
+    filteredPkResultsList[[n]] <- pkResultsDataFrame[ (pkResultsDataFrame$QuantityPath %in% op) & (pkResultsDataFrame$Parameter %in% pkParameterNamesEachOutput[[op]]), ]
+  }
+  filteredPkResultsDf <- do.call("rbind.data.frame", filteredPkResultsList)
+
+
+  filteredPkResultsDf$QuantityPath <- as.factor(filteredPkResultsDf$QuantityPath)
+  filteredPkResultsDf$Parameter <- as.factor(filteredPkResultsDf$Parameter)
+  filteredPkResultsDf$Unit <- as.factor(filteredPkResultsDf$Unit)
+  return(filteredPkResultsDf)
+}
+
+
 #' @title getSAFileIndex
 #' @description Function to build and write to CSV a dataframe that stores all
 #' sensitivity analysis result files that will be output by a population sensitivity analysis.
-#' @param pkParameterResultsFilePath path to pk parameter results CSV file
-#' @param quantileVec quantiles of distributions of pk parameter results for a population.
-#' The individual in the population that yields a pk parameter closest to the quantile is selected for sensitivity analysis.
-#' @param resultsFileFolder path to population sensitivity analysis results CSV files
+#' @param structureSet `SimulationStructure` R6 class object
+#' @param settings list of settings for the population sensitivity analysis
 #' @param resultsFileName root name of population sensitivity analysis results CSV files
-#' @param popSAResultsIndexFile name of CSV file that will store index that identifies name of
-#' sensitivity analysis results file for each sensitivity analysis run
-getSAFileIndex <- function(pkParameterResultsFilePath,
-                           pkParameterSelection,
-                           quantileVec,
-                           resultsFileName) {
-  allPKResultsDataframe <- getPKResultsDataFrame(pkParameterResultsFilePath, pkParameterSelection)
+getSAFileIndex <- function(structureSet = structureSet,
+                           settings = settings,
+                           resultsFileName = resultsFileName) {
+  quantileVec <- settings$quantileVec
+
+  allPKResultsDataframe <- getPKResultsDataFrame(structureSet)
+
   outputs <- levels(allPKResultsDataframe$QuantityPath)
-  pkParameters <- levels(allPKResultsDataframe$Parameter)
   outputColumn <- NULL
   pkParameterColumn <- NULL
   individualIdColumn <- NULL
   valuesColumn <- NULL
   unitsColumn <- NULL
   quantileColumn <- NULL
   for (output in outputs) {
+    pkParameters <- unique(allPKResultsDataframe$Parameter[allPKResultsDataframe$QuantityPath == output])
     for (pkParameter in pkParameters) {
       singleOuputSinglePKDataframe <- allPKResultsDataframe[ allPKResultsDataframe["QuantityPath"] == output & allPKResultsDataframe["Parameter"] == pkParameter, ]
       quantileResults <- getQuantileIndividualIds(singleOuputSinglePKDataframe, quantileVec)

```

Comment:
I can't select the loop below, but the lines in loop coming next can be simplified:
instead of `for(i in seq_along...`
You can directly build your data.frame: 
`saResultsByOuptut <- data.frame(Outputs = output, pkParameters = , pkParameter, 
IndividualId = quantileResults$ids, Value = quantileResults$values, ...)`
And then
`sensitivityAnalysesResultsIndexFileDF <- rbind.data.frame(sensitivityAnalysesResultsIndexFileDF,
saResultsByOuptut)`

Reply:
Fixed